docs/client-specifications.rst

```diff
@@ -0,0 +1,246 @@
+Unfortunately, this cross-platform was not leveraged by the experimentation team, which built another ad-hoc client. In front of this situation, we are obliged to admit that our idea of having a single client of reference is dead. Instead, we are going to provide specifications for Remote Settings clients, to at least mitigate the consequences of clients fragmentation.
+
+That being said, we still **strongly discourage** the implementation of ad-hoc clients.
+
+We distinguish two major use-cases:
+
+- authenticated write operations, ie. publish data;
+- anonymous read operations, ie. fetch data from within our products.
+
+Since the former does not take place on clients, it matters less than the latter, which has a major impact on traffic and our servers load.
+
+Existing Clients
+----------------
+
++------------------------+-------+-------------+--------------------------------------+---------------+----------+
+|                        | Gecko | Rust Client | application-services/remote-settings | kinto-http.py | kinto.js |
++========================+=======+=============+======================================+===============+==========+
+| Write Operations       | ❌    | ✅          | ❌                                   | ✅            | ✅       |
+| Add/Remove Attachments | ❌    | ✅          | ❌                                   | ❌            | ✅       |
+| Approve/Reject reviews | ❌    | ✅          | ❌                                   | ❌            | ❌       |
++------------------------+-------+-------------+--------------------------------------+---------------+----------+
+| Changesets Endpoints   | ✅    | ✅          | ❌                                   | ✅            | ❌       |
+| Cache Busting          | ✅    | ✅          | ❌                                   | ✅            | ❌       |
+| Env Switching          | ✅    | ~ Partial   | ✅                                   | ❌            | ❌       |
+| Signature Verification | ✅    | ✅ Optional | ❌                                   | ❌            | ❌       |
+| Local State            | ✅    | ✅ Optional | ❌                                   | ❌            | ✅       |
+| Fetch Attachments      | ✅    | ✅          | ✅ Without integrity check           | ❌            | ✅       |
+| Backoff                | ✅    | ✅          | ✅                                   | ✅            | ✅       |
+| Deprecation            | ❌    | ❌          | ❌                                   | ✅ Minimal    | ✅       |
++------------------------+-------+-------------+--------------------------------------+---------------+----------+
+
+
+Specifications
+--------------
+
+Remote Settings is a layer on top of the Kinto API. Although every read-only operation offered by Kinto is technically available on the Remote Settings server, clients must take into consideration that once ran on millions of devices, the impact on our infrastructure can be significant.
```

Comment:
Second sentence feels a little cumbersome. Maybe something like: Although every read-only operation offered by Remote Settings is available on kinto server, clients must consider that millions of devices all making the same requests could have a significant impact on infrastructure.